openvdb/points/PointConversion.h

```diff
@@ -773,19 +669,70 @@ createPointDataGrid(const PointIndexGridT& pointIndexGrid, const PositionArrayT&
 
     // create point attribute storage on each leaf
 
-    InitialiseAttributesOp<PointDataTreeT, PointIndexTreeT> initialise(
-                                pointIndexGrid.tree(), descriptor);
-    tbb::parallel_for(leafRange, initialise);
+    leafManager.foreach(
+        [&](LeafT& leaf, size_t /*idx*/) {
+
+            // obtain the PointIndexLeafNode (using the origin of the current leaf)
+            const auto* pointIndexLeaf = pointIndexTree.probeConstLeaf(leaf.origin());
+            assert(pointIndexLeaf);
+
+            // initialise the attribute storage
+            Index pointCount(static_cast<Index>(pointIndexLeaf->indices().size()));
+            leaf.initializeAttributes(descriptor, pointCount);
+        },
+    /*threaded=*/true);
 
     // populate position attribute
 
-    PopulatePositionAttributeOp<PointDataTreeT,
-                                PointIndexTreeT,
-                                PositionArrayT> populate(pointIndexTree,
-                                                        xform,
-                                                        positions);
+    const size_t positionIndex = descriptor->find("P");
+    assert(positionIndex != AttributeSet::INVALID_POS);
+
+    leafManager.foreach(
+        [&](LeafT& leaf, size_t /*idx*/) {
+
+            // obtain the PointIndexLeafNode (using the origin of the current leaf)
+            const auto* pointIndexLeaf = pointIndexTree.probeConstLeaf(leaf.origin());
+            assert(pointIndexLeaf);
+
+            auto attributeWriteHandle = AttributeWriteHandle<Vec3f, CompressionT>::create(
+                leaf.attributeArray(positionIndex));
+
```

Comment:
I wonder if we should discuss some standards for auto usage as there's none at the moment. Not going to recommend changing anything here, just seen some variable types -> auto during this change.

Reply:
I tend to try and limit it's use to specific cases - where I'm repeating the name of a type or deducing a type from a nested name, especially when it would require a typename keyword. I don't mind if we include something in our style guide.